roles/adoption_osp_deploy/tasks/config_files.yml

```diff
@@ -0,0 +1,176 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Gather ansible_user_id from undercloud
+  delegate_to: "osp-undercloud-0"
+  ansible.builtin.setup:
+    gather_subset:
+      - user_id
+      - user_dir
+
+- name: Load scenario config_download
+  vars:
+    _config_download_path: >-
+      {{
+        [cifmw_adoption_source_scenario_path,
+         _stack.config_download_file
+        ] | path_join
+      }}
+  ansible.builtin.slurp:
+    path: "{{ _config_download_path }}"
+  register: _original_config_download
+  delegate_to: "localhost"
+
+- name: Generate config download file
+  vars:
+    _config_download: >
+      {{
+        _original_config_download['content'] |
+        b64decode | from_yaml
+      }}
+    _new_config_download_fields: {}
+    _ctlplane_net: "{{ cifmw_networking_env_definition.networks.ctlplane }}"
+    _hostname_map_translation: >-
+      {{
+        cifmw_adoption_osp_deploy_scenario.hostname_groups_map |
+        ansible.utils.keep_keys(target=_stack.stack_nodes)
+      }}
+  block:
+    - name: Gather overcloud tripleo nodes
+      when: group.key is in _hostname_map_translation
+      ansible.builtin.set_fact:
+        _tripleo_nodes: >-
+          {{
+             _tripleo_nodes | default([]) +
+             group.value
+          }}
+      loop: "{{ _vm_groups | dict2items }}"
+      loop_control:
+        loop_var: group
+        label: "{{ group.key }}"
+
+    - name: Generate Hostnamemap field
+      when: group.key is in _hostname_map_translation
+      vars:
+        _tripleo_name: "{{ _hostname_map_translation[group.key] }}"
+        _group_nodes: >-
+          {%- set hosts = {} -%}
+          {%- for node in group.value -%}
+            {%- set key = _tripleo_name ~ '-' ~ loop.index0 | string  -%}
+            {%- set _ = hosts.update({key: node}) -%}
+          {%- endfor -%}
+          {{ hosts }}
+      ansible.builtin.set_fact:
+        _hostname_map: >-
+          {{
+            _hostname_map | default({}) |
+            combine(_group_nodes)
+          }}
+      loop: "{{ _vm_groups | dict2items }}"
+      loop_control:
+        loop_var: group
+        label: "{{ group.key }}"
```

Comment:
Both loops can be perfectly joined and the second one doesn't depend in the computation done in the first one.

Reply:
indeed, thanks, I've merged them